### PR TITLE
Refactor interfaces and move them into their own files and packages

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @jeromefroe 

This PR refactors the interfaces and move them into their own files and packages for more clear separation of responsibilities.

### CODE REVIEW

File: aggregator/types.go [hunk 1]
```diff
@@ -1,308 +0,0 @@
-// Copyright (c) 2016 Uber Technologies, Inc.
```
Comment (cw9) [inline aggregator/types.go]: any particular reasons to not keep the types.go pattern?

File: aggregator/handler/handler.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 Uber Technologies, Inc.
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (cw9) [inline aggregator/handler/handler.go]: rename to types.go?

File: aggregator/types.go (hunk 1 above)
Comment (xichen2020) [reply aggregator/types.go]: I feel it's easier to look at the interfaces alongside their implementations, unless there are multiple implementations of the interface, in which case it makes sense to pull out the interface into its own file. Don't think it's a conventional pattern to put all interfaces in one large `types.go` file even though they sometimes aren't closely related.

File: aggregator/handler/handler.go (hunk 2 above)
Comment (xichen2020) [reply aggregator/handler/handler.go]: There's already a `type.go` file defining different handler types hence the `handler.go`.

**Pull request merged**